### PR TITLE
jd-diff-patch: init at 1.4.0

### LINKED ISSUE #140311: `jd` - JSON diff and patch
**Project description**
jd is a commandline utility and Go library for diffing and patching JSON values.

**Metadata**

* homepage URL: https://github.com/josephburnett/jd
* source URL: https://github.com/josephburnett/jd
* license: mit
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by bryanasdev000**
###### Motivation for this change

Closes https://github.com/NixOS/nixpkgs/issues/140311 .

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blaggacao** [comment]
:+1: for the name!

I didn't realize this had a UI.

**Comment by blaggacao** [comment]
@nrdxp can you help with a merge?

File: pkgs/development/tools/jd-diff-patch/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  # not including web ui
+  excludedPackages = [ "gae" "pack" ];
+
+  vendorSha256 = "sha256-w3mFra7Je+8qIDQMSyUYucoLZ6GtrZmr56O6xmihIYc=";
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "jd is a commandline utility and Go library for diffing and patching JSON values";
```
Comment (legendofmiracles) [inline pkgs/development/tools/jd-diff-patch/default.nix:22]:
```suggestion
    description = "A commandline utility and Go library for diffing and patching JSON values";
```
Descriptions shouldn't start with the package name, and always with a uppercase letter, as by CONTRIBUTING.md:

>  * Be capitalized.
>  * Not start with the package name.
>  * Not have a period at the end.

**Review by legendofmiracles — APPROVED** [review]

**Review by superherointj — APPROVED** [review]
LGTM

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 140687 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>jd-diff-patch</li>  </ul></details>

**Pull request merged**